From 548621edcca59391e0925d002b1adfe015fc5f12 Mon Sep 17 00:00:00 2001
From: Christian Hohnstaedt <Christian.Hohnstaedt@wago.com>
Date: Wed, 11 Sep 2019 11:12:39 +0200
Subject: [PATCH] usb: gadget: Enable configfs controlled USB gadget

The CONFIG_USB_CONFIGFS driver allows to configure
the gadgets at runtime. The following gadgets are available:
 - acm (serial)
 - ncm/ecm (network Linux)
 - rndis (network windows)
 - mass_storage (USB hard-drive)

Disable TI DMA mode, which results in high CPU load.
Otherwise the [ktimersoftd/0] is running wild
and slowing down the system

Signed-off-by: Christian Hohnstaedt <Christian.Hohnstaedt@wago.com>
---
 arch/arm/configs/am335x_pac_generic_defconfig | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/arch/arm/configs/am335x_pac_generic_defconfig b/arch/arm/configs/am335x_pac_generic_defconfig
index 6ed887ce..2873bca 100644
--- a/arch/arm/configs/am335x_pac_generic_defconfig
+++ b/arch/arm/configs/am335x_pac_generic_defconfig
@@ -399,8 +399,7 @@ CONFIG_USB_MUSB_HDRC=y
 CONFIG_USB_MUSB_OMAP2PLUS=y
 CONFIG_USB_MUSB_AM35X=y
 CONFIG_USB_MUSB_DSPS=y
-CONFIG_USB_INVENTRA_DMA=y
-CONFIG_USB_TI_CPPI41_DMA=y
+CONFIG_MUSB_PIO_ONLY=y
 CONFIG_USB_SERIAL=y
 CONFIG_USB_SERIAL_QUALCOMM=y
 CONFIG_USB_SERIAL_OPTION=y
@@ -409,7 +408,12 @@ CONFIG_AM335X_PHY_USB=y
 CONFIG_USB_GPIO_VBUS=m
 CONFIG_USB_ULPI=y
 CONFIG_USB_GADGET=y
-CONFIG_USB_G_SERIAL=m
+CONFIG_USB_CONFIGFS=m
+CONFIG_USB_CONFIGFS_ACM=y
+CONFIG_USB_CONFIGFS_NCM=y
+CONFIG_USB_CONFIGFS_ECM=y
+CONFIG_USB_CONFIGFS_RNDIS=y
+CONFIG_USB_CONFIGFS_MASS_STORAGE=y
 CONFIG_MMC=y
 CONFIG_SDIO_UART=m
 CONFIG_MMC_OMAP_HS=y
@@ -430,6 +434,7 @@ CONFIG_RTC_DRV_RS5C372_IRQ_SUPPORT=y
 CONFIG_RTC_DRV_ISL1208=y
 CONFIG_DMADEVICES=y
 CONFIG_DMA_OMAP=y
+CONFIG_TI_CPPI41=y
 CONFIG_TI_EDMA=y
 CONFIG_ASYNC_TX_DMA=y
 CONFIG_DMATEST=m
-- 
2.7.4

